parlai/core/torch_agent.py

```diff
@@ -1983,6 +1986,14 @@ def batch_act(self, observations):
         # check if there are any labels available, if so we will train on them
         self.is_training = any('labels' in obs for obs in observations)
 
+        # check if we should add truncate stats
+        if all('if_text_truncate' in obs for obs in observations):
```

Comment:
I thought you'd also set one for average length pre-truncated

Also what about label truncate?

Reply:
yeah, to me that is more useful --> if every example is 1-2 tokens beyond truncation, it'd be quite misleading to see that the truncate percentage is very high, when in reality it's not nearly as destructive as thought

Reply:
label truncate is tricky to add, as I don't want to touch the return value of _vectorize_text. ```_set_label_vec``` calls '''_vectorize_text''' instead doing the truncation itself.

Reply:
There's only one re-definition of `_vectorize_text` in our entire code-base, externally and internally. I think you can upgrade its return value.

Reply:
I might misunderstood you. 
This function is called everywhere?